src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.cs

```diff
@@ -78,30 +78,22 @@ private void ProduceTags(TaggerContext<TTag> context, DocumentSnapshotSpan spanT
                 return;
             }
 
-            if (!document.TryGetText(out var sourceText))
-            {
-                return;
-            }
-
-            var editorSnapshot = sourceText.FindCorrespondingEditorTextSnapshot();
-            if (editorSnapshot == null)
-            {
-                return;
-            }
+            var editorSnapshot = spanToTag.SnapshotSpan.Snapshot;
```

Comment:
@jasonmalinowski @CyrusNajmabadi if spanToTag.SnapshotSpan.Snapshot is same as sourceText.FindCorrespondingEditorTextSnapshot, then how that ever can be failed? either 2 are different if that ever returned (TryGetText or FindCorrespoindingEditorTextSnapshot returned false or null) or there is a bug here.

we should add debug assert or NFW to make sure spanToTag.SnapshotSpan.Snapshot and document.FindCorrespondingEditorTextSnapshot returned same editorSnapshot otherwise, editorSnapshot and document is pointing to different snapshot.

Reply:
by the way, I believe existing code does exactly same thing as this? this code is just copied over from the original one.

Reply:
These have to be the same due to: https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs#L353

i.e. we only ask providers for the ITextSnapshot+TextSpan to produce tags for.  The tagger infrastructure then gets the document by calling GetOpenDocumentInCurrentContextWithChanges on that snapshot.   So the Document is always the proper one for the data we started with (unless something catastrophically bad happened in the workspace).